odl/solvers/functional/functional.py

```diff
@@ -1072,6 +1072,77 @@ def proximal(self):
         return proximal_cconj(self.convex_conj.proximal)
 
 
+def simple_functional(space, fcall=None, grad=None, prox=None, grad_lip=np.nan,
+                      cconj_fcall=None, cconj_grad=None, cconj_prox=None,
+                      cconj_grad_lip=np.nan, linear=False):
+    """Simplified interface to create a functional with specific properties.
+
+    Users may specify as many properties as is needed by the application.
+
+    Parameters
+    ----------
+    space : `LinearSpace`
+        Space that the functional should act on.
+    fcall : `callable`, optional
+        Function to evaluate when calling the functional.
+    grad : `Operator`, optional
+        Gradient operator of the functional
+    prox : `proximal factory`, optional
+        Proximal factory for the functional.
+    grad_lip : float, optional
+        lipschitz constant of the functional.
+    cconj_fcall : `callable`, optional
+        Function to evaluate when calling the convex conjugate functional.
+    cconj_grad : `Operator`, optional
+        Gradient operator of the convex conjugate functional
+    cconj_prox : `proximal factory`, optional
+        Proximal factory for the convex conjugate functional.
+    cconj_grad_lip : float, optional
+        lipschitz constant of the convex conjugate functional.
+    linear : bool, optional
+        True if the operator is linear
+    """
```

Comment:
A small doctest example would be nice, perhaps a very simple one from scratch, and one using properties of an existing functional and extending it.